### PR TITLE
Added Exception class

### PR DESCRIPTION
**Pull request by Atish-Jain**

### CODE REVIEW

File: src/main/java/com/elroykane/usermanangement/exception/NoUserFoundException.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package com.elroykane.usermanangement.exception;
+
+public class NoUserFoundException extends RuntimeException 
+{
+
+	private static final long serialVersionUID = 1L;
```
Comment (elroykanye) [inline src/main/java/com/elroykane/usermanangement/exception/NoUserFoundException.java:6]: Hey Jain. I've never really understood what serialVersionUID is for. Everytime I use eclipse, I get a warning to use for some reason🤷🏾‍♂️

File: src/main/java/com/elroykanye/userregsystem/rest/UserRegRestController.java [hunk 2]
```diff
@@ -38,31 +41,60 @@ public void setUserJpaRep(UserJpaRep userJpaRep) {
...
-        Optional<UserDTO> user = userJpaRep.findById(id);
-        return new ResponseEntity<>(user, HttpStatus.OK);
+        if(userJpaRep.findById(id).isPresent())
+        {
+        	Optional<UserDTO> user = userJpaRep.findById(id);
+            return new ResponseEntity<>(user, HttpStatus.OK);
+        }
+        else
+        {
+        	logger.error("No user found for id : "+id);
```
Comment (elroykanye) [inline src/main/java/com/elroykanye/userregsystem/rest/UserRegRestController.java:51]: I'm new to using the logger... Haven't even read the docs yet. Where can I see logs that are created by logger objects? In the project dir?

**Review by elroykanye — APPROVED** [review]
I'm so happy that you did this. Thanks a lot for the update. Hoping to collab with you more on REST projects. ✊🏾🤓

**Comment by elroykanye** [comment]
Works good

File: src/main/java/com/elroykane/usermanangement/exception/NoUserFoundException.java (hunk 1 above)
Comment (Atish-Jain) [reply src/main/java/com/elroykane/usermanangement/exception/NoUserFoundException.java:6]: I have seen this article on the javadoc: https://docs.oracle.com/javase/6/docs/api/java/io/Serializable.html
"The serialization runtime associates with each serializable class a version number, called a serialVersionUID, which is used during deserialization to verify that the sender and receiver of a serialized object have loaded classes for that object that are compatible with respect to serialization. If the receiver has loaded a class for the object that has a different serialVersionUID than that of the corresponding sender's class, then deserialization will result in an InvalidClassException. A serializable class can declare its own serialVersionUID explicitly by declaring a field named "serialVersionUID" that must be static, final, and of type long"

You can also refer: http://www.javapractices.com/topic/TopicAction.do?Id=45

File: src/main/java/com/elroykanye/userregsystem/rest/UserRegRestController.java (hunk 2 above)
Comment (Atish-Jain) [reply src/main/java/com/elroykanye/userregsystem/rest/UserRegRestController.java:51]: You can set, where you want loggers to be printed, in the loggers' XML configuration file by using "appender" tag. One can set loggers to be printed either in console or in separate .log files or maybe even in the database. One can also set the loggers print location in the combination of console, separate log files and database.

**Pull request merged**